transcom/mymove
# B-18415 GHC API Remove Now Unused Endpoints and Tech Debt

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882404&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Since we are removing the functionality of the `final_destination_address` to be updated at the service item level, and is now dependent on the shipment's `delivery_address` - the GHC API (primarily the TOO's approval/reject of a SIT address update request), do not need the following endpoints:

- `approveSITAddressUpdate`
- `rejectSITAddressUpdate`
- `createSITAddressUpdate`
- `updateMTOServiceItem` - `PATCH` request (this was not being used anywhere in the UI or backend)

This PR does the following:
- removed all backend use and tech debt of the `approveSITAddressUpdate` endpoint
- removed all backend use and tech debt of the `rejectSITAddressUpdate` endpoint
- removed all backend use and tech debt of the `createSITAddressUpdate` endpoint
- removed tech debt of unused endpoint `updateMTOServiceItem`
- removed existing mocks related to above endpoints

**NOTE** I am leaving the `approveSITAddressUpdate`, `rejectSITAddressUpdate`, and `createSITAddressUpdate` occurrences in the `MoveHistory` folder. This is because they are being used in staging since this functionality has been previously used and tested.

### How to test

1. Access MM as an office user
2. Login as a TOO
3. Try and update a service item's destination address... anywhere
4. You can also check out [Swagger GHC UI](http://officelocal:3000/swagger-ui/ghc.html#/) and see the above endpoints are now removed
5. Another reference would be `swagger-def` - all occurrences of above endpoints should no longer be there
